### DISCUSSION TITLE
View options for entire chain

### DISCUSSION THREAD

**Discussion by felixdo-vorwerk** [jobbflykt/x509-mode#36] [General]
Hei, first of all, thanks for this wonderful package.
I often have to look a chain of certs, i.e. a file with a sequence of BEGIN CERTIFICATE/END CERTIFICATE sections. Do you have some helpers or functions to x509-dwim the entire chain?

**Comment by jobbflykt** [comment]
It's possible to add `x509-dwim-{next-prev}` functions that look for an intersting adjacent region in the source buffer and displays that.  Could be bound to `n` and `p` in `x509-mode` and `x509-asn1-mode`. Sounds like a useful function. I will look into it.

**Comment by jobbflykt** [comment]
The commit c3874db7001bc2726b1548d05950671dd052ed24 adds `x509-dwim-next` and `x509-dwim-prev` bound to `n` and `p` in `x509-mode` and `x509-asn1-mode`. This is not quite was was asked for but may still be useful in some usecases.
Parsing and processing all regions in a buffer, e.g. a chain of certificates, and sending the output to the same buffer will require some new infrastructure in x509-mode. I will think about how that can be implemented.

**Comment by jobbflykt** [comment]
b30bc5e91b28ade6802d80513954a7ec4f025db2 adds `x509-swoop` that can be used to view all certificates (or any PKI data) in a file. Hopefully this is useful @felixdo-vorwerk
